perun/postprocess/moving_average/run.py

```diff
@@ -0,0 +1,223 @@
+"""
+Postprocessor module with non-parametric analysis using the moving average methods.
+"""
+
+import click
+import functools
+
+import perun.logic.runner as runner
+import perun.postprocess.moving_average.methods as methods
+import perun.postprocess.regression_analysis.data_provider as data_provider
+import perun.postprocess.regression_analysis.tools as tools
+import perun.utils.cli_helpers as cli_helpers
+from perun.utils.helpers import PostprocessStatus, pass_profile
+
+__author__ = 'Simon Stupinsky'
+
+# set the labels at the center of the window as default
+_DEFAULT_CENTER = True
+# default computational method - Simple Moving Average
+_DEFAULT_MOVING_METHOD = 'sma'
+# specify decay in terms of Center of Mass (com) as default
+_DEFAULT_DECAY = ('com', 0)
+# default statistic function to compute - mean/average
+_DEFAULT_STATISTIC = 'mean'
+# recognized window types for Simple Moving Average/Median
+_WINDOW_TYPES = ['boxcar', 'triang', 'blackman', 'hamming', 'bartlett', 'parzen', 'bohman', 'blackmanharris', 'nuttall',
+                 'barthann']
+
+
+# TODO: The possibility of before postprocessing phase
+
+def postprocess(profile, **configuration):
+    """
+    Invoked from perun core, handles the postprocess actions
+
+    :param dict profile: the profile to analyze
+    :param configuration: the perun and options context
+    """
+    # Perform the non-parametric analysis using the moving average methods
+    moving_average_models = methods.compute_moving_average(data_provider.data_provider_mapper(profile, **configuration),
+                                                           configuration)
+
+    # Return the profile after the execution of moving average method
+    return PostprocessStatus.OK, '', {'profile': tools.add_models_to_profile(profile, moving_average_models)}
+
+
+# TODO: The possibility of after postprocessing phase
+
+
+def common_options(f):
```

Comment:
Very strong suggestion: While this seems like good hack, I can't say I particularly like this solution. Consider moving this to the group instead and propagate through context.

The difference is in how the perun will be executed from cli:
1) perun postprocessby <postprocess_by_options> moving_average ema <common_opt> <ema_specific>
2) perun postprocessby <postprocess_by_options> moving_average <common_opt> ema <ema_specific>

I would say (2) is better and can be solved in Click (check other CLI functions, I'm sure I have this behaviour along with passing context solved already). Or maybe, screw this, finish the MA and I will refactor this myself (could be faster).